### ISSUE TITLE
Content Assist fails "The pipe is being closed"

### ISSUE THREAD

**Issue by Pingger** [eclipse/corrosion#330]
Hi,

I'm currently getting into Rust, it seems, that I have encountered an issue the Eclipse extension for it.
The Content assist lists only the option "Error while computing completion"
![image](https://user-images.githubusercontent.com/4761298/90200969-c3275d00-ddd9-11ea-82b8-5c2ee0a8c04f.png)
![image](https://user-images.githubusercontent.com/4761298/90200998-d63a2d00-ddd9-11ea-8322-9c55d42ae30a.png)

Writing the code manually correctly highlights the errors and warnings in the code, The Content assist also sometimes suggests some templates (e.g. for `for`-Loops).
Compiling and running is also no issue.

The full exception from the error log is this:
<details>
<summary>StackTrace</summary>

```
java.util.concurrent.ExecutionException: org.eclipse.lsp4j.jsonrpc.JsonRpcException: java.io.IOException: The pipe is being closed
	at java.util.concurrent.CompletableFuture.reportGet(Unknown Source)
	at java.util.concurrent.CompletableFuture.get(Unknown Source)
	at org.eclipse.lsp4e.operations.completion.LSContentAssistProcessor.computeCompletionProposals(LSContentAssistProcessor.java:105)
	at org.eclipse.jface.text.contentassist.AsyncCompletionProposalPopup.lambda$10(AsyncCompletionProposalPopup.java:351)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.jface.text.contentassist.AsyncCompletionProposalPopup.lambda$9(AsyncCompletionProposalPopup.java:350)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source)
	at java.util.concurrent.CompletableFuture$AsyncSupply.exec(Unknown Source)
	at java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(Unknown Source)
	at java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
	at java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
Caused by: org.eclipse.lsp4j.jsonrpc.JsonRpcException: java.io.IOException: The pipe is being closed
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageConsumer.consume(StreamMessageConsumer.java:72)
	at org.eclipse.lsp4e.LanguageServerWrapper.lambda$1(LanguageServerWrapper.java:243)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.request(RemoteEndpoint.java:161)
	at org.eclipse.lsp4j.jsonrpc.services.EndpointProxy.invoke(EndpointProxy.java:91)
	at com.sun.proxy.$Proxy26.completion(Unknown Source)
	at org.eclipse.lsp4e.operations.completion.LSContentAssistProcessor.lambda$1(LSContentAssistProcessor.java:101)
	at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(Unknown Source)
	at java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.util.stream.AbstractPipeline.evaluateToArrayNode(Unknown Source)
	at java.util.stream.ReferencePipeline.toArray(Unknown Source)
	at org.eclipse.lsp4e.operations.completion.LSContentAssistProcessor.lambda$0(LSContentAssistProcessor.java:104)
	at java.util.concurrent.CompletableFuture.uniCompose(Unknown Source)
	at java.util.concurrent.CompletableFuture$UniCompose.tryFire(Unknown Source)
	at java.util.concurrent.CompletableFuture$Completion.exec(Unknown Source)
	... 4 more
Caused by: java.io.IOException: The pipe is being closed
	at java.io.FileOutputStream.writeBytes(Native Method)
	at java.io.FileOutputStream.write(Unknown Source)
	at java.io.BufferedOutputStream.flushBuffer(Unknown Source)
	at java.io.BufferedOutputStream.write(Unknown Source)
	at java.io.FilterOutputStream.write(Unknown Source)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageConsumer.consume(StreamMessageConsumer.java:67)
	... 20 more
```

</details>

Im using the eclipse version from the eclipse.org Website: https://www.eclipse.org/downloads/packages/ "Eclipse IDE for Rust Developers" 2020-06-R on Windows x64

With this Version (I installed the Spectrum theme from the Marketplace):
<details>
<summary>Eclipse IDE Installation Details</summary>

```
Corrosion: Rust edition in Eclipse IDE	0.4.2.201908051347	org.eclipse.corrosion.feature.feature.group	Eclipse Corrosion
Eclipse IDE for Rust Developers	4.16.0.20200615-1200	epp.package.rust	Eclipse Packaging Project
Git integration for Eclipse	5.8.0.202006091008-r	org.eclipse.egit.feature.group	Eclipse EGit
Git integration for Eclipse - Gitflow support	5.8.0.202006091008-r	org.eclipse.egit.gitflow.feature.feature.group	Eclipse EGit
Marketplace Client	1.8.3.v20200526-1554	org.eclipse.epp.mpc.feature.group	Eclipse Marketplace Client
Mylyn WikiText	3.0.36.202002070035	org.eclipse.mylyn.wikitext_feature.feature.group	Eclipse Mylyn
Spectrum Theme	1.0.0.202008050400	com.aobuchow.themes.spectrum.feature.feature.group	aobuchow
TM Terminal	4.6.0.202001311822	org.eclipse.tm.terminal.feature.feature.group	Eclipse CDT
```

</details>

**Comment by mickaelistria** [comment]
Looking at the stack, can you please check in Windows > Preferences > Language Servers > Log that logging is __disabled__ for your language servers? The stack seems to say that logging is active and the issue could be the file being externally modified.
Can you please share the version of Rust and RLS you have installed locally? Also, are you running the IDE with recent version of Java?
Do you have an example of a file that systematically triggers that? Does a plain hello world fail for you when querying completion?

**Comment by Pingger** [comment]
![image](https://user-images.githubusercontent.com/4761298/92972002-c3228780-f481-11ea-912f-e51f8fac5605.png)
Looks disabled to me...
RLS Version: rls 1.41.0 (085f24b 2020-05-28)
Rust Version: rustc 1.45.2 (d3fb005a3 2020-07-31)
Java version was at the time latest openjdk-14.0.1 (now it is 14.0.2

I have no file that systematically triggers it, it just always shows the aforementioned exception

**Comment by mickaelistria** [comment]
So I just tried all that on my Windows VM and unfortunately(?) I cannot reproduce and all works like a charm. The only diff I see is that I'm running `rustc 1.46.0`.

**Comment by mickaelistria** [comment]
As I cannot reproduce it, I won't be able to help. Can you please try upgrading to newer rustc, and can you please then enable the logs for RLS via the page you shown earlier, and then share those logs upon failure?

**Comment by Pingger** [comment]
Hmm,

apparently not touching rust for a month also solved the issue ...
I didn't even need to update anything ... It just started working again ...
I also enabled logging, so incase it popsup again I can provide the logs.

I am currently updating rustc, but as I mentioned it started working before updating ...
Closing the issue for now in hoping it doesn't pop up again. (Will reopen if it does)

Greetings and Thanks for trying to reproduce it.
